### PR TITLE
SchedulerV3: basic scheduler don't add tables to the stopping capture

### LINKED ISSUE #6112: SchedulerV3: draining capture should give up if it becomes the owner.
Deploy 2 captures, `capture-0` and `capture-1`, and `capture-0` is the owner.

Drain the `capture-1` moves all tables on it to the `capture-0`, now use `systemctl stop` to let owner crash, only one capture `capture-1` alive now, and it becomes the owner.

all old owner `capture-0` tables will be added on `capture-1`, since `capture-1` is `stopping` now, it rejects add table request, then causes all replication tasks cannot make progress.

```
[2022/06/29 11:35:39.931 +08:00] [INFO] [agent.go:267] ["tpscheduler: agent is stopping, and decline handle add table request"] [capture=b0ccb623-8b87-40f7-9184-1e983277bc14] [namespace=default] [changefeed=blackhole-test] [request="add_table:<table_id:329 is_secondary:true checkpoint:<checkpoint_ts:434234648111874128 resolved_ts:434234648111874128 > > "]
[2022/06/29 11:35:39.931 +08:00] [INFO] [agent.go:267] ["tpscheduler: agent is stopping, and decline handle add table request"] [capture=b0ccb623-8b87-40f7-9184-1e983277bc14] [namespace=default] [changefeed=blackhole-test] [request="add_table:<table_id:338 is_secondary:true checkpoint:<checkpoint_ts:434234648111874128 resolved_ts:434234648111874128 > > "]
[2022/06/29 11:35:39.931 +08:00] [INFO] [agent.go:267] ["tpscheduler: agent is stopping, and decline handle add table request"] [capture=b0ccb623-8b87-40f7-9184-1e983277bc14] [namespace=default] [changefeed=blackhole-test] [request="add_table:<table_id:342 is_secondary:true checkpoint:<checkpoint_ts:434234648111874128 resolved_ts:434234648111874128 > > "]
[2022/06/29 11:35:39.931 +08:00] [INFO] [agent.go:267] ["tpscheduler: agent is stopping, and decline handle add table request"] [capture=b0ccb623-8b87-40f7-9184-1e983277bc14] [namespace=default] [changefeed=blackhole-test] [request="add_table:<table_id:346 is_secondary:true checkpoint:<checkpoint_ts:434234648111874128 resolved_ts:434234648111874128 > > "]
[2022/06/29 11:35:39.931 +08:00] [INFO] [agent.go:267] ["tpscheduler: agent is stopping, and decline handle add table request"] [capture=b0ccb623-8b87-40f7-9184-1e983277bc14] [namespace=default] [changefeed=mysql-test] [request="add_table:<table_id:150 is_secondary:true checkpoint:<checkpoint_ts:434234647587324106 resolved_ts:434234647587324106 > > "]
```

Solution:

* when the agent receives a heartbeat from the new owner, clear the `stopping`
* owner should not schedule `add table` or `move table` request to the `stopping` capture.

* In the 2 captures deployment, if the owner crash, the draining capture should give up to make sure all tables can be in replicating.
* If there are more than 2 captures, such as 3 living. The owner crash may let any one of the other 2 captures become the new owner, all its tables may be redistributed to any capture.
  * the new owner should not distribute tables to a `stopping` capture
  * the `stopping` capture should give up the draining process if it's the only alive capture.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #6112 

#### Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/scheduler/internal/tp/scheduler_basic.go [hunk 1]
```diff
@@ -74,9 +74,26 @@ func (b *basicScheduler) Schedule(
 	// Build add table tasks.
 	if len(newTables) > 0 {
 		captureIDs := make([]model.CaptureID, 0, len(captures))
-		for captureID := range captures {
+		for captureID, status := range captures {
+			if status.State == CaptureStateStopping {
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/scheduler_basic.go:78]: Good catch, please add some tests to cover it.

File: cdc/scheduler/internal/tp/agent.go [hunk 2]
```diff
@@ -35,25 +35,47 @@ import (
...
 	// Liveness of the capture.
 	//
-	// CaptureLivenessShuttingDown rejects all add table requests.
+	// LivenessCaptureStopping rejects all add table requests.
 	liveness model.Liveness
 }
 
+func (a agentInfo) String() string {
+	bytes, _ := json.Marshal(a)
+	return string(bytes)
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/agent.go:60]: What does it look like? I think json skips unexported fileds, so it's blank in this case.

File: cdc/scheduler/internal/tp/scheduler_basic_test.go [hunk 3]
```diff
@@ -31,7 +31,16 @@ func TestSchedulerBasic(t *testing.T) {
 	// AddTable only
 	replications := map[model.TableID]*ReplicationSet{}
 	b := newBasicScheduler()
+
+	// all capture's stopping, cannot add table
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/scheduler_basic_test.go:35]: Could you test only one capture is stopping?

**Review by overvenus — COMMENTED** [review]
The PR title does not fit its change, please update the title.

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**